qiskit_experiments/library/characterization/t1_analysis.py

```diff
@@ -135,19 +135,22 @@ def fit_fun(x, a, tau, c):
         bounds = {"a": amplitude_bounds, "tau": t1_bounds, "c": offset_bounds}
         fit_result = curve_fit(fit_fun, xdata, ydata, init, sigma=sigma, bounds=bounds)
 
-        result_data = {
-            "value": fit_result["popt"][1],
-            "stderr": fit_result["popt_err"][1],
-            "unit": "s",
-            "fit": fit_result,
-            "quality": self._fit_quality(
+        result = AnalysisResultData(
+            result_data = {"value": fit_result["popt"][1],
```

Comment:
I was also thinking to define something like (mainly for printing purpose)
```python
@dataclasses.dataclass(frozen=True)
class FitValue:
    name: str
    value: float
    stdev: Optional[float] = None
    unit: Optional[str] = None

    def __str__(self):
        if self.stdev is not None:
            value = f"{self.value}\u00B1{self.stdev}"
        else:
            value = f"{self.value}"
        if self.unit:
            return f"{self.name}={value} [{self.unit}]"
        else:
            return f"{self.name}={value}"
```
This is extension of float value and can be added to JSON serializer.

Reply:
That looks nice, @nkanazawa1989